lib/pbench/server/database/models/datasets.py

```diff
@@ -366,6 +370,27 @@ def validate_access(self, key: str, value: str) -> str:
             return access
         raise DatasetBadParameterType(value, "access keyword")
 
+    @validates("owner")
+    def validate_owner(self, key: str, value: Any) -> User:
+        """
+        Validate and translate owner name to User object
+        Args:
+            key: owner
+            value: username
+        Raises:
+            DatasetBadParameter: the owner value given doesn't resolve to a
+                Pbench username.
+        Returns:
+            User object
+        """
+        if type(value) is User:
+            return value
+        elif type(value) is str:
+            user = User.query(username=value)
+            if user:
+                return user
+        raise DatasetBadParameterType(value, "username")
+
```

Comment:
Actually, that's a good catch, I didnt update the dataset creation in the upload, prior to making this change when the user_id was just a string without any foreign key constraint I used the `owner_id` to populate the dataset owner_id field. With the new users table now I should really assign an `owner` value instead of just `owner_id` and let the sqlalchemy handle the mapping of the foreign key.

Reply:
> I'm feeling like maybe this code is evidence of a dog-wagging tail...maybe we should be removing the _need_ for this code rather than supporting it.

Probably a separate issue to track down.

Reply:
> With the new users table now I should really assign an `owner` value instead of just `owner_id`

Why?  On upload, we have a token, which yields a UUID, which is what we use to label ownership of the dataset...why bring the `Users` table into creating an entry in the `Datasets` table?

Reply:
Yeah, so I did some experiments with the database and it seems like if we just keep the `user_id` column (foreign key) and remove the SQLAlchemy relationship then this wouldn't make much difference from the SQL operations. So I can probably remove his validation function as we wouldn't be using it anywhere to create a Dataset. But I'll defer to @dbutenhof to confirm if we should remove the SQLAlchemy relationship.

Reply:
Okay, actually there is one issue in removing the SQLAlchemy relationship, consider the following:
If I do 
```
class Dataset(Base):
    owner_id = Column(String, ForeignKey("users.id"), nullable=False)
    owner = relationship("User")
    resource_id = Column(String(255), unique=True, nullable=False)
    name = Column(String(1024), unique=False, nullable=False)

######################
# Case 1
ds = Dataset(owner_id="nonexistent", resource_id="deed1esd", name="fio")
ds.add()

print(ds.owner) # Prints None

######################
# Case 2
user = User(
            id="12345",
            username="username",
        )
ds = Dataset(owner=user, resource_id="deed1esd", name="fio")
ds.add()
print(ds.owner) # prints the User object
assert User.query(username="drb_new") # creates a new user object in the database
```
case 1, sqlalchemy doesn't check if the foreign key exists or not, it just creates the column and if we reference the owner later it returns None.
Case 2, Sqlalchemy actually creates a new User entry in our table if we specify the `owner` column instead of the owner_id, and the `owner_id` column also gets correctly populated with the user's id.

For the above reason, I think it's worth keeping the relationship and assigning `owner` value instead of `owner_id` column when creating a Dataset in the upload (Even though we should already have a User entry in our Users table before we create a Dataset in the upload). But I don't have a strong opinion on this one.